torchgeo/datasets/substation_seg.py

```diff
@@ -0,0 +1,170 @@
+            # stds = np.array([157, 254, 290, 420, 363, 457, 575, 606, 630, 156, 3, 554, 523]).reshape(-1, 1, 1)
+            image = (image - self.args.means) / self.args.stds
+        else:
+            image = image / self.normalizing_factor
+            # clipping image to 0,1 range
+            image = np.clip(image, 0, 1)
+
+        # selecting channels
+        if self.in_channels == 3:
+            image = image[:, [3, 2, 1], :, :]
+        else:
+            if self.model_type == 'swin':
+                image = image[
+                    :, [3, 2, 1, 4, 5, 6, 7, 10, 11], :, :
+                ]  # swin only takes 9 channels
+            else:
+                image = image[:, : self.in_channels, :, :]
+
+        # handling multiple images across timepoints
+        if self.use_timepoints:
+            image = image[:4, :, :, :]
+            if self.args.timepoint_aggregation == 'concat':
+                image = np.reshape(
+                    image, (-1, image.shape[2], image.shape[3])
+                )  # (4*channels,h,w)
+            elif self.args.timepoint_aggregation == 'median':
+                image = np.median(image, axis=0)
+        else:
+            # image = np.median(image, axis=0)
+            # image = image[0]
+            if self.args.timepoint_aggregation == 'first':
+                image = image[0]
+            elif self.args.timepoint_aggregation == 'random':
+                image = image[np.random.randint(image.shape[0])]
+
+        mask = np.load(mask_path)['arr_0']
+        mask[mask != 3] = 0
+        mask[mask == 3] = 1
+
+        image = torch.from_numpy(image)
+        mask = torch.from_numpy(mask).float()
+        mask = mask.unsqueeze(dim=0)
+
+        if self.mask_2d:
+            mask_0 = 1.0 - mask
+            mask = torch.concat([mask_0, mask], dim=0)
+
+        if self.image_resize:
+            image = self.image_resize(image)
+
+        if self.mask_resize:
+            mask = self.mask_resize(mask)
+
+        return {'image': image, 'mask': mask}
+
+    def __len__(self) -> int:
+        """Returns the number of items in the dataset."""
+        return len(self.image_filenames)
+
+    def plot(self) -> None:
+        """Plots a random image and mask from the dataset."""
+        index = np.random.randint(0, self.__len__())
+        data = self.__getitem__(index)
+        image = data['image']
+        mask = data['mask']
+
+        fig, axs = plt.subplots(1, 2, figsize=(15, 15))
+        axs[0].imshow(image.permute(1, 2, 0).cpu().numpy())
+        axs[1].imshow(image.permute(1, 2, 0).cpu().numpy())
+        axs[1].imshow(mask.squeeze().cpu().numpy(), alpha=0.5, cmap='gray')
+
+    def _verify(self) -> None:
+        """Checks if dataset exists, otherwise download it."""
+        image_dir_exists = os.path.exists(self.image_dir)
+        mask_dir_exists = os.path.exists(self.mask_dir)
+        if not (image_dir_exists and mask_dir_exists):
+            self._download()
```

Comment:
Downloading random files from the internet without checksum verification should not happen by default, the user should have to pass `download=True` if they really want things to be downloaded. This violates the principle of least surprise.